samples/common/gnss_nmea.c

```diff
@@ -111,10 +111,7 @@ int nmea_handler(xbee_dev_t *xbee, const void FAR *raw,
     	payload_len = sizeof(nmea_str) - 1;
     }
 
-    memcpy(nmea_str, &gnss->payload[0], payload_len);
-    nmea_str[payload_len] = '\0';
-
-    printf("%s\n", nmea_str);
+    printf("%.*s\n", payload_len, gnss->payload);
```

Comment:
You can also delete local variable `nmea_str` and the size check on `payload_len`.